navit/maptool/buffer.c

```diff
@@ -54,16 +71,24 @@ load_buffer(char *filename, struct buffer *b, long long offset, long long size)
 	}
 	b->size=b->malloced=size;
 	dbg_assert(b->size>0);
-#if 0
-	fprintf(stderr,"reading "LONGLONG_FMT" bytes from %s of "LONGLONG_FMT" bytes at "LONGLONG_FMT"\n", b->size, filename, len, offset);
-#endif
+	
 	fseeko(f, offset, SEEK_SET);
 	b->base=malloc(b->size);
 	dbg_assert(b->base != NULL);
-	fread(b->base, b->size, 1, f);
+	if (fread(b->base, b->size, 1, f) == 0){
+		dbg(lvl_warning, "fread failed");
+		return 0;
+        }
 	fclose(f);
+	return 1;
 }
-
+/**
+ * @brief Determines size of buffer for file 
+ *
+ * This function determines the size of the buffer required to read a file. 
+ *
+ * @param  filename Name of file for which the required size of the buffer is determined
+ */
```

Comment:
Can you please document the return value too?

Reply:
Can you please remind me wheat it returns? Don't know the function ftello.